Included net-5.0 specific updates for Modern Web Applications with ASP.NET Core - eBook

sughosneo opened the pull request:
This PR updates following pages :

- `Characteristics of modern web applications`
- `Choose Between Traditional Web Apps and Single Page Apps (SPAs)`
- `Architectural principles`
- `Azure hosting recommendations for ASP.NET Core web apps`
- `Common web application architectures`
- `Common client-side web technologies`
- `Developing ASP.NET Core MVC apps`
- `Development process for Azure`
- `Working with Data in ASP.NET Core Apps`
- `Test ASP.NET Core MVC apps`

sughosneo committed:
Included net-5.0 specific updates

sughosneo committed:
Fixed lint error in common-client-side page.

sughosneo commented:
Please note, Index page is not yet modified. It will get updated once the other changes are approved.

sughosneo committed:
Removed minor version from the dotnet version.

ardalis commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -154,23 +154,33 @@ This is enough to render "Hello Vue!" on the page. Note, however, that Vue isn't
+
+Unlike other JavaScript frameworks, `Blazor WebAssembly` is a single-page app (SPA) framework for building interactive client-side web apps with .NET. Blazor WebAssembly uses open web standards without plugins or recompiling code into other languages. Blazor WebAssembly works in all modern web browsers, including mobile browsers.
+
+Running .NET code inside web browsers is made possible by WebAssembly (abbreviated `wasm`). WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed. WebAssembly is an open web standard and supported in web browsers without plugins.
```
```suggestion
Running .NET code inside web browsers is made possible by WebAssembly (abbreviated `wasm`). WebAssembly is a compact bytecode format optimized for fast download and maximum execution speed. WebAssembly is an open web standard and is supported in web browsers without plugins.
```

ardalis commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -154,23 +154,33 @@ This is enough to render "Hello Vue!" on the page. Note, however, that Vue isn't
 - How opinionated is the framework, and do you agree with its default way of doing things?
 
-- Does it (or a companion library) include all of the features your app requires?
+- Does it (or a companion library) includes all of the features your app requires?
```
I wouldn't make this change. "Does it includes all of the features" doesn't make sense to me.

IEvangelist replied:
I agree with @ardalis here, I'd leave it as "include"
```suggestion
- Does it (or a companion library) include all of the features your app requires?
```

ardalis commented on docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md:
```diff
@@ -57,11 +57,11 @@ Many web applications involve a combination of traditional web application behav
 ## Simple development and deployment
 
-ASP.NET Core applications can be written using simple text editors and command-line interfaces, or full-featured development environments like Visual Studio. Monolithic applications are typically deployed to a single endpoint. Deployments can easily be automated to occur as part of a continuous integration (CI) and continuous delivery (CD) pipeline. In addition to traditional CI/CD tools, Microso ...
+ASP.NET Core applications can be written using simple text editors and command-line interfaces, or full-featured development environments like Visual Studio. Monolithic applications are typically deployed to a single endpoint. Deployments can easily be automated to occur as part of continuous integration (CI) and continuous delivery (CD) pipeline. In addition to traditional CI/CD tools, Microsoft ...
```
This change only works if you get rid of "pipeline" at the end. Deployments can be part of CI or part of CD but they need to be part of *a* CI or CD *pipeline*.

sughosneo replied:
I see. Made the changes again.

ardalis commented on docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md:
```diff
@@ -57,11 +57,11 @@ Many web applications involve a combination of traditional web application behav
 ## Traditional ASP.NET and Web Forms
 
-In addition to ASP.NET Core, traditional ASP.NET 4.x continues to be a robust and reliable platform for building web applications. ASP.NET supports MVC and Web API development models, as well as Web Forms, which is well suited to rich page-based application development and features a rich third-party component ecosystem. Microsoft Azure has great longstanding support for ASP.NET 4.x applications, ...
+In addition to ASP.NET Core, traditional ASP.NET 4.x continues to be a robust and reliable platform for building web applications. ASP.NET supports MVC and Web API development models, as well as Web Forms, which are well suited to rich page-based application development and features a rich third-party component ecosystem. Microsoft Azure has great longstanding support for ASP.NET 4.x applications ...
```
If we're going to say "Web Apps for Containers" is singular, then the product "Web Forms" is also singular.

sughosneo replied:
Yes, that makes sense. Reverted back the change.

ardalis reviewed (changes requested):
A few minor changes requested, thanks!

Commit:
Update docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md

Co-authored-by: Steve Smith <steve@kentsmiths.com>

Commit:
Update docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md

Co-authored-by: David Pine <david.pine@microsoft.com>

sughosneo committed:
Included review feedback

sughosneo committed:
Updated index page

sughosneo commented:
> A few minor changes requested, thanks!

Thanks @ardalis  for your feedback. 
I have incorporated all the suggestions and also updated the `index` page for further review.
